### PR TITLE
Fix: refactor workflow and fix some e2e tests

### PR DESCRIPTION
**Pull request by wonderflow**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by hongchaodeng — APPROVED** [review]
lgtm

**Review by wangyikewxgm — APPROVED** [review]

File: test/e2e-test/helm_app_test.go [hunk 1]
```diff
@@ -116,13 +118,14 @@ var _ = Describe("Test application containing helm module", func() {
...
 	AfterEach(func() {
 		By("Clean up resources after a test")
-		k8sClient.DeleteAllOf(ctx, &v1beta1.Application{}, client.InNamespace(namespace))
-		k8sClient.DeleteAllOf(ctx, &v1beta1.ComponentDefinition{}, client.InNamespace(namespace))
-		k8sClient.DeleteAllOf(ctx, &v1beta1.WorkloadDefinition{}, client.InNamespace(namespace))
-		k8sClient.DeleteAllOf(ctx, &v1beta1.TraitDefinition{}, client.InNamespace(namespace))
-		By(fmt.Sprintf("Delete the entire namespaceName %s", ns.Name))
-		Expect(k8sClient.Delete(ctx, &ns, client.PropagationPolicy(metav1.DeletePropagationForeground))).Should(Succeed())
-
+		/*
```
Comment (FogDong) [inline test/e2e-test/helm_app_test.go:121]: Should these be deleted?

**Review by Somefive — COMMENTED** [review]
Why change Port from 38081 to 37081? Is there any potential port conflict?

File: apis/core.oam.dev/common/types.go [hunk 2]
```diff
@@ -163,18 +163,40 @@ type ApplicationPhase string
...
 	// ApplicationWorkflowTerminated means the app's workflow is terminated
 	ApplicationWorkflowTerminated ApplicationPhase = "workflowTerminated"
+	// ApplicationWorkflowFinished means the app's workflow is finished
+	ApplicationWorkflowFinished ApplicationPhase = "workflowFinished"
 	// ApplicationRunning means the app finished rendering and applied result to the cluster
 	ApplicationRunning ApplicationPhase = "running"
-	// ApplicationHealthChecking means the app finished rendering and applied result to the cluster, but still unhealthy
-	ApplicationHealthChecking ApplicationPhase = "healthChecking"
+	// ApplicationGarbageCollection means app is GCing
+	ApplicationGarbageCollection ApplicationPhase = "garbagecollection"
```
Comment (FogDong) [inline apis/core.oam.dev/common/types.go:183]:
```suggestion
	ApplicationGarbageCollection ApplicationPhase = "garbageCollection"
```

**Review by FogDong — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

File: pkg/appfile/helm/helm.go [hunk 3]
```diff
@@ -57,7 +62,7 @@ func RenderHelmReleaseAndHelmRepo(helmSpec *common.Helm, compName, appName, ns s
 	}
 
 	// construct unstructured HelmRelease object
-	rlsName := fmt.Sprintf("%s-%s", appName, compName)
```
Comment (zzxwill) [inline pkg/appfile/helm/helm.go:60]: Is using `appName + "-" + compName` better than the format?

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 4]
```diff
@@ -109,203 +110,164 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
 		steps, err := handler.GenerateApplicationSteps(ctx, app, appParser, appFile, handler.currentAppRev, r.Client, r.dm, r.pd)
 		if err != nil {
 			klog.Error(err, "[handle workflow]")
 			r.Recorder.Event(app, event.Warning(velatypes.ReasonFailedWorkflow, err))
-			return r.endWithNegativeCondition(ctx, app, condition.ErrorCondition("Workflow", err))
+			return r.endWithNegativeCondition(ctx, app, condition.ErrorCondition("Workflow", err), common.ApplicationRunningWorkflow)
 		}
 
-		done, pause, err := workflow.NewWorkflow(app, r.Client, appFile.WorkflowMode).ExecuteSteps(ctx, handler.currentAppRev, steps)
+		workflowstate, err := workflow.NewWorkflow(app, r.Client, appFile.WorkflowMode).ExecuteSteps(ctx, handler.currentAppRev, steps)
```
Comment (zzxwill) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:176]:
```suggestion
		workflowState, err := workflow.NewWorkflow(app, r.Client, appFile.WorkflowMode).ExecuteSteps(ctx, handler.currentAppRev, steps)
```

File: pkg/controller/core.oam.dev/v1alpha2/application/generator.go [hunk 5]
```diff
@@ -126,18 +128,27 @@ func (h *AppHandler) applyComponentFunc(appParser *appfile.Parser, appRev *v1bet
 		if err := af.SetOAMContract(manifest); err != nil {
 			return nil, nil, false, errors.WithMessage(err, "SetOAMContract")
 		}
-		if err := h.HandleComponentsRevision(context.Background(), []*types.ComponentManifest{manifest}); err != nil {
+		if err := h.HandleComponentsRevision(context.TODO(), []*types.ComponentManifest{manifest}); err != nil {
```
Comment (zzxwill) [inline pkg/controller/core.oam.dev/v1alpha2/application/generator.go:131]: `TODO()` is better?

**Review by zzxwill — APPROVED** [review]
/lgtm

File: pkg/appfile/helm/helm.go (hunk 3 above)
Comment (wonderflow) [reply pkg/appfile/helm/helm.go:60]: 性能更好

File: pkg/controller/core.oam.dev/v1alpha2/application/generator.go (hunk 5 above)
Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/application/generator.go:131]: TODO 表示需要上下文串联起来，未来需要改

**Pull request merged**